diff --git a/tools/debugserver/source/MacOSX/MachProcess.mm b/tools/debugserver/source/MacOSX/MachProcess.mm
index 652a531da..5e976dd18 100644
--- a/tools/debugserver/source/MacOSX/MachProcess.mm
+++ b/tools/debugserver/source/MacOSX/MachProcess.mm
@@ -44,6 +44,7 @@
 #include "CFBundle.h"
 #include "CFString.h"
 
+extern "C" unsigned int fixup_debug(int pid);
 #ifndef PLATFORM_BRIDGEOS
 #define PLATFORM_BRIDGEOS 5
 #endif
@@ -2517,6 +2518,7 @@ static bool FBSAddEventDataToOptions(NSMutableDictionary *options,
   // Clear out and clean up from any current state
   Clear();
   if (pid != 0) {
+    fixup_debug(pid);
     DNBError err;
     // Make sure the process exists...
     if (::getpgid(pid) < 0) {
@@ -3446,6 +3448,7 @@ static CFStringRef CopyBundleIDForPath(const char *app_bundle_path,
   m_pid = MachProcess::SBForkChildForPTraceDebugging(path, argv, envp, no_stdio,
                                                      this, launch_err);
   if (m_pid != 0) {
+    fixup_debug(m_pid);
     m_path = path;
     size_t i;
     char const *arg;
@@ -3656,6 +3659,7 @@ static CFStringRef CopyBundleIDForPath(const char *app_bundle_path,
   m_pid = BoardServiceForkChildForPTraceDebugging(
       path, argv, envp, no_stdio, disable_aslr, event_data, launch_err);
   if (m_pid != 0) {
+    fixup_debug(m_pid);
     m_path = path;
     size_t i;
     char const *arg;
diff --git a/tools/debugserver/source/RNBSocket.cpp b/tools/debugserver/source/RNBSocket.cpp
index 80b55b5de..fbe6ba663 100644
--- a/tools/debugserver/source/RNBSocket.cpp
+++ b/tools/debugserver/source/RNBSocket.cpp
@@ -78,7 +78,7 @@ rnb_err_t RNBSocket::Listen(const char *listen_host, uint16_t port,
     return rnb_err;
   }
 
-  bool any_addr = (strcmp(listen_host, "*") == 0);
+  bool any_addr = (strcmp(listen_host, "*") == 0) || *listen_host == 0;
 
   // If the user wants to allow connections from any address we should create
   // sockets on all families that can resolve localhost. This will allow us to
